lms/djangoapps/ccx/views.py

```diff
@@ -0,0 +1,526 @@
+        })
+
+
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@coach_dashboard
+def ccx_grades_csv(request, course):
+    """
+    Download grades as CSV.
+    """
+    # Need course module for overrides to function properly
+    field_data_cache = FieldDataCache.cache_for_descriptor_descendents(
+        course.id, request.user, course, depth=2)
+    course = get_module_for_descriptor(
+        request.user, request, course, field_data_cache, course.id)
+    ccx = get_ccx_for_coach(course, request.user)
+    with ccx_context(ccx):
+        # The grading policy for the MOOC is probably already cached.  We need
+        # to make sure we have the CCX grading policy loaded.
+        course._field_data_cache = {}  # pylint: disable=protected-access
+        course.set_grading_policy(course.grading_policy)
+
+        enrolled_students = User.objects.filter(
+            ccxmembership__ccx=ccx,
+            ccxmembership__active=1
+        ).order_by('username').select_related("profile")
+        grades = iterate_grades_for(course, enrolled_students)
+
+        header = None
+        rows = []
+        for student, gradeset, __ in grades:
+            if gradeset:
+                # We were able to successfully grade this student for this
+                # course.
+                if not header:
+                    # Encode the header row in utf-8 encoding in case there are
+                    # unicode characters
+                    header = [section['label'].encode('utf-8')
+                              for section in gradeset[u'section_breakdown']]
+                    rows.append(["id", "email", "username", "grade"] + header)
+
+                percents = {
+                    section['label']: section.get('percent', 0.0)
+                    for section in gradeset[u'section_breakdown']
+                    if 'label' in section
+                }
+
+                row_percents = [percents.get(label, 0.0) for label in header]
+                rows.append([student.id, student.email, student.username,
+                             gradeset['percent']] + row_percents)
+
+        buf = StringIO()
+        writer = csv.writer(buf)
+        for row in rows:
+            writer.writerow(row)
+
+        return HttpResponse(buf.getvalue(), content_type='text/plain')
+
+
+@login_required
+def switch_active_ccx(request, course_id, ccx_id=None):
+    """set the active CCX for the logged-in user
+    """
+    user = request.user
+    course_key = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+    # will raise Http404 if course_id is bad
+    course = get_course_by_id(course_key)
+    course_url = reverse(
+        'course_root', args=[course.id.to_deprecated_string()]
+    )
+    if ccx_id is not None:
+        try:
+            requested_ccx = CustomCourseForEdX.objects.get(pk=ccx_id)
+            assert requested_ccx.course_id.to_deprecated_string() == course_id
```

Comment:
There are other places where SlashSeparatedCourseKey is used in this manner. Do you recommend to use CourseKey there as well?